File: 02-frontend/react-records/src/models/Record.ts

class Record { 

   id: number;
   title: string;
   artist?: string;
   description?: string;
   copies?: number;
   copiesAvailable?: number;
   genre?: string;
   img?: string

   constructor(id: number, title: string, artist: string, description: string, copies: number, copiesAvailable: number, genre: string, img: string){
      this.id = id;
      this.title = title;
      this.artist = artist;
      this.description = description;
      this.copies = copies;
      this.copiesAvailable = copiesAvailable;
      this.genre = genre;
      this.img = img;
   }

}

export default Record;

File: 01-backend/spring-boot-records/src/main/java/com/records/springbootrecords/entity/Record.java
package com.records.springbootrecords.entity;

import lombok.Data;

import javax.persistence.*;

@Entity
@Data
@Table(name = "record")
public class Record {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private long id;

    @Column(name = "title")
    private String title;

    @Column(name = "artist")
    private String artist;

    @Column(name = "description")
    private String description;

    @Column(name = "copies")
    private int copies;

    @Column(name = "copies_available")
    private int copiesAvailable;

    @Column(name= "genre")
    private String genre;

    @Column(name = "img")
    private String img;
}
